source/uwp/Renderer/lib/XamlBuilder.cpp

```diff
@@ -2768,6 +2750,120 @@ AdaptiveNamespaceStart
         AddInputValueToContext(renderContext, adaptiveCardElement, *toggleInputControl);
     }
 
+    _Use_decl_annotations_
+    void XamlBuilder::BuildMedia(
+        IAdaptiveCardElement* adaptiveCardElement,
+        IAdaptiveRenderContext* renderContext,
+        IAdaptiveRenderArgs* renderArgs,
+        IUIElement** mediaControl)
+    {
+        ComPtr<IAdaptiveCardElement> localCardElement{ adaptiveCardElement };
+        ComPtr<IAdaptiveMedia> adaptiveMedia;
+        THROW_IF_FAILED(localCardElement.As(&adaptiveMedia));
+
+        ComPtr<IAdaptiveHostConfig> hostConfig;
+        THROW_IF_FAILED(renderContext->get_HostConfig(&hostConfig));
+
+        // Get the poster image
+        ComPtr<IImage> posterImage;
+        GetMediaPosterAsImage(renderContext, renderArgs, adaptiveMedia.Get(), &posterImage);
+
+        // If the host doesn't support interactivity we're done here, just return the poster image
+        if (!SupportsInteractivity(hostConfig.Get()))
+        {
+            renderContext->AddWarning(
+                ABI::AdaptiveNamespace::WarningStatusCode::InteractivityNotSupported,
+                HStringReference(L"Media was present in card, but interactivity is not supported").Get());
+
+            THROW_IF_FAILED(posterImage.CopyTo(&mediaControl));
+            return;
```

Comment:
If there's no poster image, on interactive cards we'll use the media element, but if interactivity is not supported, we'll fail?